### PR TITLE
Refactoring/accountless client

### PR DESCRIPTION
**Pull request by EvgenKor**
1. Client library interaction approach was changed (supported multipool and accountless configuration):
2. `.env` file was removed. The configuration currently located in [client-config.json](https://github.com/zkBob/zkbob-console/blob/9140e0acc0c527e643ceb5840a6434151d9192d4/client-config.json) file
3. Added ability to interact with relayer without zkAccount (one-time inform placeholder) 
image
4. You can switch between pools when you is login in you account: 
image
5. You can get balance and redeem your giftcard (giftcard spending key and birth index are required) 
image
6. `gift-cards` command was renamed to `gift-card-generate` to avoid confusion

The console should built with the associated client library branch [refactoring/accountless-client](https://github.com/zkBob/zkbob-client-js/tree/refactoring/accountless-client)

Linked PR: https://github.com/zkBob/zkbob-client-js/pull/112

### CODE REVIEW

**Comment by EvgenKor** [comment]
The default console configuration supports Sepolia and Goerli pools

File: src/account.ts [hunk 1]
```diff
@@ -332,72 +477,83 @@ export default class Account {
...
+        const curChainId = chainId ?? env.pools[this.getCurrentPool()].chainId;
+        const txUrl = env.blockExplorerUrls[String(curChainId)].tx;
+        return txUrl.replace('{{hash}}', txHash);
     }
 
-    public getTransactionUrl(txHash: string): string {
-        return this.client.getTransactionUrl(txHash);
+    public getAddressUrl(addr: string, chainId: number | undefined = undefined): string {
+        const curChainId = chainId ?? env.pools[this.getCurrentPool()].chainId;
+        const txUrl = env.blockExplorerUrls[String(curChainId)].tx;
```
Comment (AllFi) [inline src/account.ts]:
```suggestion
        const txUrl = env.blockExplorerUrls[String(curChainId)].address;
```

File: src/account.ts [hunk 2]
```diff
@@ -332,72 +477,83 @@ export default class Account {
     }
 
     public async transferToken(to: string, amount: bigint): Promise<string> {
-        return await this.client.transferToken(TOKEN_ADDRESS, to, amount.toString());
+        const tokenAddress = env.pools[this.getCurrentPool()].tokenAddress;
```
Comment (AllFi) [inline src/account.ts]:
```suggestion

```

File: src/account.ts [hunk 3]
```diff
@@ -548,67 +698,80 @@ export default class Account {
...
-        try {
-            seed = AES.decrypt(cipherText, password).toString(Utf8);
-            if (!bip39.validateMnemonic(seed)) throw new Error('invalid mnemonic');
-        } catch (_) {
-            throw new Error('Incorrect password');
-        }
-
-        return seed;
+    public async proverVersion(): Promise<ServiceVersion> {
+        this.assertZpClient();
```
Comment (AllFi) [inline src/account.ts]: Maybe it makes sense to add function `zpClient()` with checks from assertZpClient. This way we can just write something like `return await this.zpClient().getProverVersion();`

File: src/account.ts [hunk 4]
```diff
@@ -1,32 +1,22 @@
...
 import bip39 from 'bip39-light';
 import HDWalletProvider from '@truffle/hdwallet-provider';
-import { deriveSpendingKeyZkBob } from 'zkbob-client-js/lib/utils';
-import { NetworkType } from 'zkbob-client-js/lib/network-type';
-import { EvmNetwork } from 'zkbob-client-js/lib/networks/evm';
-import { PolkadotNetwork } from 'zkbob-client-js/lib/networks/polkadot';
-
-import { TransferRequest } from 'zkbob-client-js/lib/client';
-import { ProverMode } from 'zkbob-client-js/lib/config';
+import Web3 from 'web3'
```
Comment (AllFi) [inline src/account.ts]:
```suggestion

```

File: src/commands.ts [hunk 5]
```diff
@@ -1,28 +1,63 @@
...
+
+    this.pause();
+    await this.account.switchPool(poolAlias, password);
+    this.resume();
+    this.echo(`Current pool: ${await this.account.getCurrentPool()}`);
+}
 
 export async function setSeed(seed: string, password: string) {
-    await this.account.login(seed, password);
+    //await this.account.login(seed, password);
```
Comment (AllFi) [inline src/commands.ts]: I guess we can just remove the whole function.

**Comment by AllFi** [comment]
@EvgenKor can you merge https://github.com/zkBob/zkbob-console/pull/65 please? In the new cloud, this request requires an authorization header.

File: src/commands.ts (hunk 5 above)
Comment (EvgenKor) [reply src/commands.ts]: Removed in https://github.com/zkBob/zkbob-console/pull/63/commits/039e8a8a86e788daa8464cb657922321045dcb70

File: src/account.ts (hunk 3 above)
Comment (EvgenKor) [reply src/account.ts]: Of course! Probably I made so in a hurry :joy:

Reworked in https://github.com/zkBob/zkbob-console/pull/63/commits/039e8a8a86e788daa8464cb657922321045dcb70

File: src/index.ts [hunk 6]
```diff
@@ -267,20 +271,26 @@ jQuery(async function ($) {
               throw new Error('Password is too weak');
             }
 
+            this.echo(`Creating new account...`);
+
             this.pause();
-            await this.account.init(seed, password, isNewAccount, initLibCallback);
+            await this.account.attachAccount(accountName, seed, password, true);
```
Comment (AllFi) [inline src/index.ts]: It doesn't look right to me because I can't restore my account state if I've cleaned the browser store.

Comment (AllFi) [reply src/index.ts]: I guess we should return back isNewAccount and related logic.

Comment (EvgenKor) [reply src/index.ts]: Fixed in https://github.com/zkBob/zkbob-console/pull/63/commits/f0cb9cddfc1ebb0c0277284a12638156bf009e44

**Review by AllFi — APPROVED** [review]
It works properly and looks good to me.

**Review by r0wdy1 — APPROVED** [review]

File: webpack.config.prod.js [hunk 7]
```diff
@@ -18,5 +18,5 @@ module.exports = merge(common, {
             }
         ],
     },
-    plugins: [new CompressionPlugin({ exclude: 'env.js' })]
+    plugins: [/*new CompressionPlugin({ exclude: 'env.js' })*/s]
```
Comment (AllFi) [inline webpack.config.prod.js]:
```suggestion
    plugins: [/*new CompressionPlugin({ exclude: 'env.js' })*/]
```

Comment (EvgenKor) [reply webpack.config.prod.js]: Fixed in https://github.com/zkBob/zkbob-console/pull/63/commits/06cb8e97f53c54135a07964854232a1bf0dfe2f2

**Pull request merged**